src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -452,3 +472,27 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         )
         df._reader = self
         return df
+
+    def _upload_local_file_to_stage(self, path: str, format: str) -> str:
+
+        if not path.startswith(STAGE_PREFIX) and not os.path.exists(path):
+            raise ValueError(f"The local file {path} does not exist")
+        temp_stage = self._session.get_session_stage()
```

Comment:
Once a file is uploaded and read into a table, it becomes unuseful (trash in the session stage). 
If `session.read` read data from the same stage path the next time, will it read the trash files so the result data could be wrong? 
In a production env, a session may be alive for long time. All these trash will be kept. So they should be dropped after they're used. To avoid this, I suggest:
1. Use a random prefix after the session stage each time when this API is called.
2. Remove the trashes by the end of read operations.

Reply:
No, if no option is specifies in `sesison.read`, we will use select query from the file directly, e.g.:
```
SELECT $1::INT AS "A", $2::STRING AS "B", $3::DOUBLE AS "C" FROM @SNOWPARK_TEMP_STAGE_BF5CP972WY/testCSV.csv( FILE_FORMAT  => '"TESTDB_SNOWPARK_PYTHON"."TESTSCHEMA_SNOWPARK_PYTHON".SNOWPARK_TEMP_FILE_FORMAT_KW11239MUH')
```
If we don't cache the result first, we will always need this file on the stage. 

Also, if the users want to remove this file after uploading, they can set `PURGE` to true in options. https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#copy-options-copyoptions

Reply:
For user's own files on stage, users can decide to use "PURGE" or not. For these auto-uploaded files for loading into a table, I feel it's better that they're automatically purged by default without user intervention.